src/irmin-pack/irmin_pack.ml

```diff
@@ -399,6 +399,52 @@ struct
     end
   end
 
+  let integrity_check (t : X.Repo.t) =
+    Fmt.pr "running the integrity check\n%!";
```

Comment:
instead of using `stdout` we should be passing a `ppf` argument to that function and use `Fmt.pf ppf` instead. It will make clear in the type that there is some printing done as a side of calling that function.